### PR TITLE
[FIX] website_forum: limit portal/user rights on forum.post.vote

### PR DESCRIPTION
**Pull request by rdeodoo**
Create and write rights are granted on portal and user.
We need to improve the security.

task-1934286

task-1862656

opw-1862218
opw-1862111
opw-1857912
opw-1861426

Github-24986
forum-135070

### CODE REVIEW

**Comment by rdeodoo** [comment]
@JKE-be 
As this fix seems to do the job on our own forum (local diff), what do you suggest?
This could be merged in earlier version where the bug is (at least 11.0?).
Let me know, I will rebase and aim good version.

**Comment by Yenthe666** [comment]
V11 pretty please, it is a bug after all. ;)
Thanks again for the good fix and communication @rdeodoo

**Comment by JKE-be** [comment]
but yes, need to be done in version where bug exists

**Comment by rdeodoo** [comment]
@JKE-be Should this become a priority?
Would be fast, just need to backport it to first impacted version.
Sad to see that PR not merged :p

**Comment by rdeodoo** [comment]
It will be rebased on 12.1 and merged ASAP

File: addons/website_forum/models/forum.py [hunk 1]
```diff
@@ -860,46 +864,66 @@ def _get_karma_value(self, old_vote, new_vote, up_karma, down_karma):
 
     @api.model
     def create(self, vals):
+        # can't modify owner of a vote
+        if self._uid != SUPERUSER_ID:
```
Comment (JKE-be) [inline addons/website_forum/models/forum.py]: are you sure, since we add support of superuser and admin ?

**Comment by JKE-be** [comment]
robodoo delegate @rdeodoo

File: addons/website_forum/models/forum.py (hunk 1 above)
Comment (rdeodoo) [reply addons/website_forum/models/forum.py]: Indeed, good catch, I simply rebased on 12.1 but this had to be replaced by `_is_admin()`.
Note that other occurence of SUPERUSER_ID wrongly appear in website_forum (I added the fix in this PR and I will backport it + fix others (website_slides) at some point).

**Comment by JKE-be** [comment]
robodoo r+

**Comment by rdeodoo** [comment]
Merged in saas-12.1 at https://github.com/odoo/odoo/commit/1dac41313005c9035a5f8d3c9db83dcad7883f24

**Pull request closed without merging**